backend/app/repositories/search_profile_repository.py

```diff
@@ -1,113 +1,151 @@
+        response_profiles.append(
+            SearchProfileDetailResponse(
+                id=profile.id,
+                name=profile.name,
+                organization_emails=organization_emails,
+                profile_emails=profile_emails,
+                public=profile.is_public,
+                editable=is_editable,
+                is_editable=is_editable,
+                owner=profile.created_by_id,
+                is_owner=is_owner,
+                topics=topic_responses,
             )
-
-            if organization_id is not None:
-                stmt = stmt.where(
-                    (
-                        (SearchProfile.is_public.is_(True))
-                        & (SearchProfile.organization_id == organization_id)
-                    )
-                    | (SearchProfile.users.any(id=user_id))
-                )
-            else:
-                stmt = stmt.where(SearchProfile.users.any(id=user_id))
-
-            result = await session.exec(stmt)
-            return result.all()
-
-    @staticmethod
-    async def update_by_id(
-        profile_id: UUID, data: SearchProfileUpdateRequest
-    ) -> Type[SearchProfile] | None:
-        async with async_session() as session:
-            profile = await session.get(SearchProfile, profile_id)
-            if not profile:
-                return None
-
-            # Update simple fields
-            profile.name = data.name
-            profile.is_public = data.public
-            profile.is_editable = data.is_editable
-            profile.owner_id = data.owner
-            profile.is_owner = data.is_owner
-
-            # Update email lists
-            if data.organization_emails is not None:
-                profile.organization_emails = data.organization_emails
-
-            if data.profile_emails is not None:
-                profile.profile_emails = data.profile_emails
-
-            # Update subscriptions (replace with new list)
-            if data.subscriptions is not None:
-                profile.subscriptions.clear()
-                for sub_data in data.subscriptions:
-                    profile.subscriptions.append(sub_data.to_model())
-
-            # Update topics (replace with new list)
-            if data.topics is not None:
-                profile.topics.clear()
-                for topic_data in data.topics:
-                    profile.topics.append(topic_data.to_model())
-
-            session.add(profile)
-            await session.commit()
-            await session.refresh(profile)
-
-            return profile
+        )
+
+    return response_profiles
+
+
+async def update_by_id(
+    profile_id: UUID, data: SearchProfileUpdateRequest
+) -> Type[SearchProfile] | None:
+    async with async_session() as session:
+        profile = await session.get(SearchProfile, profile_id)
+        if not profile:
+            return None
+
+        # Update simple fields
+        profile.name = data.name
+        profile.is_public = data.public
```

Comment:
```suggestion
        profile.is_public = data.is_public
```

why this inconsistency?